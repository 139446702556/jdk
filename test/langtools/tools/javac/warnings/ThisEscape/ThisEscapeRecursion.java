/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @compile/ref=ThisEscapeRecursion.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeRecursion.java
 * @summary Verify 'this' escape detection properly handles leaks via recursive methods
 */

public class ThisEscapeRecursion {

    public ThisEscapeRecursion() {
        this.noLeak(0);         // no leak here
        this.mightLeak();       // possible leak here
    }

    public final void noLeak(int depth) {
        if (depth < 10)
            this.noLeak(depth - 1);
    }

    public void mightLeak() {
    }
}
