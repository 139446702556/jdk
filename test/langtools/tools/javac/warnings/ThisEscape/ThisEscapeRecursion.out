ThisEscapeRecursion.java:12:23: compiler.warn.possible.this.escape
1 warning
